android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1700,6 +1703,17 @@ public void onMyPositionModeChanged(int newMode)
     if (controller.isPlanning())
       showAddStartOrFinishFrame(controller, true);
     LocationHelper.from(this).restartWithNewMode();
+    switch (newMode) {
+    case PENDING_POSITION:
+      Utils.keepScreenOn(true, getWindow());
+      break;
```

Comment:
So, is it possible if a position is not available, we will get an infinite screen here with "Keep searching" alert?

Reply:
It is... Probably it wasn't a good idea. Instead, we start a foreground service with a notification to show that the location search is in progress. I added an alternative solution for PENDING_POSITION (https://github.com/organicmaps/organicmaps/pull/6436) and probably can add a foreground service & notification later. This is out of scope of this PR and I am removing keepScreenOn() for PENDING_POSITION from the code.

Reply:
Removed the wake lock for PENDING_POSITION mode.